pypeit/extraction.py

```diff
@@ -317,8 +313,12 @@ def extract(self, global_sky, model_noise=None, spat_pix=None):
         # Return
         return self.skymodel, self.objmodel, self.ivarmodel, self.outmask, self.sobjs
 
-    def prepare_extraction(self):
+    def prepare_extraction(self, global_sky):
```

Comment:
This method is going to disappear in my JWST branch I'm a bit worried about conflicts there that I cannot easily resolve. Let's discuss how to proceed here. This PR precedes that one so should go forward, but maybe you can think about how you are going to restructure what you are doing here in that context.